polaris-react/src/components/IndexTable/IndexTable.scss

```diff
@@ -218,6 +218,7 @@ $loading-panel-height: 53px;
   font-size: var(--p-font-size-75);
   white-space: nowrap;
   border: 0;
+  z-index: var(--p-z-1);
```

Comment:
We recently added this fix in Web by targeting the Polaris heading CSS property (https://github.com/Shopify/web/pull/83341). This fix is needed in this PR as the underlined tooltips are showing above the sticky columns.